test/core/TakerOperations.t.sol

```diff
@@ -1019,6 +1019,35 @@ contract TakerOperationsTest is MangroveTest {
     assertEq(otherTkrBalanceBaseBefore, base.balanceOf($(otherTkr)), "other taker's base balance should not change");
     assertEq(otherTkrBalanceQuoteBefore, quote.balanceOf($(otherTkr)), "other taker's quote balance should not change");
   }
+
+  function test_unconsumed_tick_leaves_correct_leaf_start_at_tick_leave_one_only(
+    int24 _tick,
+    bool crossTick,
+    bool leaveOneOnly
+  ) public {
+    quote.approve($(mgv), 10_000 ether);
+    _tick = int24(bound(_tick, -100, 100));
+    int24 _firstPostedTick = crossTick ? _tick - 1 : _tick;
+    mkr.newOfferByTick(_firstPostedTick, 1 ether, 100_000);
+    mkr.newOfferByTick(_tick, 1 ether, 100_000);
+    uint ofr3 = mkr.newOfferByTick(_tick, 1 ether, 100_000);
+    uint ofr4 = mkr.newOfferByTick(_tick, 1 ether, 100_000);
+    uint volume = leaveOneOnly ? 3 ether : 2 ether;
+    mgv.marketOrderByTick($(base), $(quote), _tick, volume, true);
+
+    Tick tick = Tick.wrap(_tick);
+
+    uint bestId = leaveOneOnly ? ofr4 : ofr3;
+    MgvStructs.OfferPacked best = pair.offers(bestId);
+    Leaf leaf = pair.leafs(best.tick().leafIndex());
+    assertEq(leaf.firstOfIndex(tick.posInLeaf()), bestId, "wrong first of tick");
+    assertEq(leaf.lastOfIndex(tick.posInLeaf()), ofr4, "wrong last of tick");
+    assertEq(pair.local().tickPosInLeaf(), tick.posInLeaf(), "wrong local.tickPosInleaf");
+    assertEq(best.prev(), 0, "best.prev should be 0");
```

Comment:
These assertions seem like something we'd like in more places ("assert that this is the best tick and that the first and last offers in it are X and Y"), so maybe we could make it an `assert*` function?